hie.yaml

```diff
@@ -1,6 +1,17 @@
 cradle:
-  cabal:
-    - path: "./src"
-      component: "lib:jrec"
-    - path: "./test"
-      component: "test:jrec-test"
+  multi:
+    - path: ./Setup.hs
+      config:
+        cradle:
+          direct:
+            arguments:
+              - "-package Cabal"
+              - "-package base"
+    - path: ./
+      config:
+        cradle:
+          cabal:
+            - path: "./src"
+              component: "lib:jrec"
+            - path: "./test"
+              component: "test:jrec-test"
```

Comment:
You can simplify by this by getting rid of `Setup.hs` (not really required?) and using:

```yaml
cradle:
  cabal:
```

https://github.com/srid/ema/blob/master/hie.yaml

Reply:
You also need `other-modules` https://gist.github.com/srid/27237dfd02e64150e5fa563c2ffff590

Reply:
i have removed `Setup.hs`. It is used for building and packaging, but we are using nix. So you are right, it is not required.